src/main/java/javaslang/collection/List.java

```diff
@@ -426,7 +426,15 @@ default int lastIndexOf(T element) {
     @Override
     default Tuple2<List<T>, List<T>> partition(Predicate<? super T> predicate) {
         Objects.requireNonNull(predicate, "predicate is null");
-        return Tuple.of(filter(predicate), filter(predicate.negate()));
+        List<T> left = nil(), right = nil();
+        for (T t : this) {
+            if (predicate.test(t)) {
+                left = left.prepend(t);
+            } else {
+                right = right.prepend(t);
+            }
+        }
+        return Tuple.of(left.reverse(), right.reverse());
```

Comment:
Mmhhh, nope, that's not right what I wrote. An ArrayList could be iterated backward to create a javaslang List, i.e. there is no need to call reverse(). That would be 1 pass in sum for partition(). However, I need to provide a better implementation for List.ofAll().

Thanks Sebastian, I will do that!